target/scripts/helpers/accounts.sh

```diff
@@ -164,3 +165,46 @@ function _create_dovecot_alias_dummy_accounts
     done < /tmp/docker-mailserver/postfix-virtual.cf
   fi
 }
+
+function _create_masters
+{
+  : >"${DOVECOT_MASTERDB_FILE}"
+
+  if [[ -f /tmp/docker-mailserver/dovecot-masters.cf ]]
```

Comment:
According the relevant [Dovecot docs](https://doc.dovecot.org/configuration_manual/authentication/master_users/), this may be an issue for LDAP users requiring a config update on their end (or rather us to add it with our other LDAP script changes in `setup-stack.sh`:

> `result_success=continue` doesn’t work with PAM or LDAP without `auth_bind=yes`, because both of them require knowing the user’s password.

That could be added to ``target/dovecot/auth-master.inc` at the end of [`_setup_ldap`](https://github.com/docker-mailserver/docker-mailserver/blob/3cb5668b6f62258d8c21629c0f5e649085e4f0b6/target/scripts/startup/setup-stack.sh#L308) like similar logic checking a file exists to add a modification.

Reply:
Hi @polarathene,

Have you had a chance to review my suggestion above?

Reply:
> Have you had a chance to review my suggestion above?

Apologies, busy week! :sweat_smile: 

---

> Do you mean something like the following in `_setup_ldap`:

<details>
<summary>Original suggestion (Slight misunderstanding)</summary>
I don't think you need to check if the file exists, unless the other maintainers think it's important to log a warning. We have quite a few `sed` operations that don't do this, so it'd seem most suitable here?:

https://github.com/docker-mailserver/docker-mailserver/blob/3cb5668b6f62258d8c21629c0f5e649085e4f0b6/target/scripts/startup/setup-stack.sh#L374-L375

```bash
sed -i -r '/!include auth-ldap\.conf\.ext/s/^#//' /etc/dovecot/conf.d/10-auth.conf
sed -i -r '/!include auth-passwdfile\.inc/s/^/#/' /etc/dovecot/conf.d/10-auth.conf

# Support Dovecot master user: https://doc.dovecot.org/configuration_manual/authentication/master_users/
# > `result_success=continue` doesn’t work with PAM or LDAP without `auth_bind=yes`, because both of them require knowing the user’s password.
sed -i -r '/auth_bind = yes/s/^#//' /etc/dovecot/conf.d/auth-master.inc
```

Optionally if using indentation before `#`, this will only remove the `#` (change to `-r` is required here unless escaping the `(` and `)`):
`sed -i -r '/auth_bind = yes/s/^(\s*)#/\1/' /etc/dovecot/conf.d/auth-master.inc`

</details>

Yes, but I don't think we need to log a warning for this file? So just conditionally run `sed` if it exists:

```bash
  if [[ -f /etc/dovecot/conf.d/auth-master.inc ]]
  then
    # Support Dovecot master user: https://doc.dovecot.org/configuration_manual/authentication/master_users/
    # > `result_success=continue` doesn’t work with PAM or LDAP without `auth_bind=yes`, because both of them require knowing the user’s password.
    sed -i -r '/auth_bind = yes/s/^#//' /etc/dovecot/conf.d/auth-master.inc
  fi
```

---

> I can't really try this properly as I don't have LDAP setup. Do you want me to commit this?

Yes. I don't think there should be any issues with it. Our LDAP support isn't well tested/supported due to no maintainers using it. Still I rather have some feature consistency/parity in place where possible instead of drift.

I've been pressed for time, but in future if I manage to refactor the test suite to be more flexible, it should be possible in future to run most of the test suite under different setups such as LDAP. If the related tests do have an issue or a user raises one before then, we should be able to troubleshoot back to this PR and opt to fix it or drop support with LDAP.

Reply:
Thanks for your comments; I wasn't chasing you, just not sure if I should have @ mentioned you in my initial comment!

I've committed your suggested change, so I will resolve this conversation now.